### PR TITLE
Adapt for v0.40.0

### PR DESCRIPTION
**Pull request by sirouk**

### CODE REVIEW

**Comment by mateuszjasiuk** [comment]
@sirouk Thank you for the contribution. So the problem you are having is due to serde not knowing how to serialize the `Transfer` struct. 
```rs
pub struct Transfer {
    /// Sources of this transfer
    pub sources: BTreeMap<Account, DenominatedAmount>,
    /// Targets of this transfer
    pub targets: BTreeMap<Account, DenominatedAmount>,
    /// Hash of tx section that contains the MASP transaction
    pub shielded_section_hash: Option<TxId>,
}
```
More precisely it can't deserialize ` BTreeMap<Account, DenominatedAmount>,` as the key of this map is a struct and JSON does not allow objects as a key of another object. So solution is to write custom serialization/deserialization functions. Unfortunately we can't implement those functions directly for `BTreeMap<Account, DenominatedAmount>` so we have to wrap it in a new type. You can take a look at this gist https://gist.github.com/mateuszjasiuk/144e590d1a5fabe79bb72b85fbe469a7 I hope this helps!

**Comment by mateuszjasiuk** [comment]
@sirouk Hello 👋 , will you have time to work on that? If not I can add fix on top of your changes and we could merge evrything in.

**Comment by sirouk** [comment]
> @sirouk Hello 👋 , will you have time to work on that? If not I can add fix on top of your changes and we could merge evrything in.

I’m away for one more day, so if you beat me to it that would be great!

**Comment by joel-u410** [comment]
👋 Hello,

I have also started a branch to update for `v0.40.0` and will push it shortly. I didn't realize this one was already open when I started mine. I expect to have a new PR shortly, and plan to address the serialization for `Transfers` there :)

(cc @mateuszjasiuk and @sirouk)

File: shared/src/block.rs [hunk 1]
```diff
@@ -283,28 +283,33 @@ impl Block {
...
+    
+                            // Iterate over sources in the BTreeMap
+                            for (source, _amount) in transfer_data.sources.iter() {
+                                let transfer_source = Id::from(source.owner.clone());
+                                changes.push(BalanceChange::new(
                                     transfer_source,
-                                    transfer_token.clone(),
-                                ),
-                                BalanceChange::new(
+                                    native_token.clone(),
```
Comment (joel-u410) [inline shared/src/block.rs:295]: One quick note here -- I don't think this is necessarily correct. The `transfer_data.sources` contains `Account` types which have `owner` and `token` defined, so this should be `source.token` not `native_token`, I think.

**Comment by joel-u410** [comment]
> 👋 Hello,
> 
> I have also started a branch to update for `v0.40.0` and will push it shortly. I didn't realize this one was already open when I started mine. I expect to have a new PR shortly, and plan to address the serialization for `Transfers` there :)
> 
> (cc @mateuszjasiuk and @sirouk)

This is done now in https://github.com/anoma/namada-indexer/pull/80

**Comment by mateuszjasiuk** [comment]
Closed in favor of #80. Thank you 🙏

**Pull request closed without merging**